semgrep/semgrep/semgrep_main.py

```diff
@@ -193,7 +193,13 @@ def main(
         lambda rule: rule.mode == JOIN_MODE,
         filtered_rules,
     )
-    filtered_rules = rest_of_the_rules
+    dependency_aware_rules, _ = partition(
```

Comment:
I also worry about this complexity & maintenance cost. Can we downscope this to require all depends-on rules to be runnable?

Reply:
Not without moving the work into the core, which I don't think makes sense -- see comment on https://github.com/returntocorp/semgrep/pull/4357#discussion_r765170208

Reply:
Is that a reason we can't move this code into the core eventually?

Reply:
after discussion with @emjin and @nbrahms, we agreed we will eventually move this into the core. But for now it will be handled in python. In a later commit, I removed the partition so the same rules are going to the core, and are just post-processed in Python, which should make it a bit cleaner when it is time to do it all in the core.